### PR TITLE
Add @__DIR__ macro.

### PR DESCRIPTION
**Pull request by helgee**
Another nightly PR but this time with a branch and all...

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -1463,4 +1463,9 @@ if VERSION < v"0.6.0-dev.374"
     end
 end
 
+if VERSION < v"0.6.0-dev.404"
```
Comment (tkelman) [inline src/Compat.jl]: did you run the commit-name script at the merge commit?

Comment (TotalVerb) [inline src/Compat.jl]: This should probably be `!isdefined(Base, Symbol("@__DIR__"))` for clarity

Comment (helgee) [inline src/Compat.jl]: @tkelman yes. Since CI says that it was wrong, how do I do it correctly? Or should I use @TotalVerb's suggestion?

Comment (tkelman) [inline src/Compat.jl]: `isdefined` sounds fine to me. Suggests there may be a bug in that script though...

Comment (tkelman) [inline src/Compat.jl]: hm, I get
```
$ git checkout -q 11ff9abd013e8c8f745616057c1076e6c6b761c5 && contrib/commit-name.sh
0.6.0-dev.528
```

Comment (helgee) [inline src/Compat.jl]: Obviously I did not run it at the merge commit. Sorry, my bad!

**Comment by helgee** [comment]
With the correct version CI passes 👍

File: src/Compat.jl (hunk 1 above)
Comment (yuyichao) [inline src/Compat.jl]: https://github.com/JuliaLang/Compat.jl/pull/281#discussion_r77931523 is still better

**Pull request merged**